src/main/scala/com/databricks/spark/avro/ConfluentSparkAvroUtils.scala

```diff
@@ -60,12 +66,59 @@ class ConfluentSparkAvroUtils(schemaRegistryURLs: String) extends Serializable {
 
   @transient private lazy val deserializer = new KafkaAvroDeserializer(schemaRegistry)
 
+  @transient private lazy val AES_MAGIC_BYTES = Set[Byte](2, 3)
+
+  @transient private lazy val awsKms: AWSKMS = new AWSKMSClient()
+
+  @transient private lazy val keyCache = MutableMap[Seq[Byte], Array[Byte]]()
+
+  val HEADER_SIZE = 5
+  val AES_ENCRYPTED_KEY_SIZE = 184
+
   def deserializerForSubject(subject: String, version: String): UserDefinedFunction = {
     udf(
       (payload: Array[Byte]) => {
```

Comment:
Make this a separated function so it can be tested (today or in the future), without a dataframe.

Reply:
the function is not context free, so code readability may suffer, prefer to postpone this